mewarpx/mewarpx/diags_store/particle_histogram_diagnostic.py

```diff
@@ -145,37 +109,126 @@ def get_fileprefix(self, species):
+                 linres_y=30, plot=True, **kwargs):
+        """Initialize diagnostic.
+
+        Arguments:
+            diag_steps (int): Number of steps between each output.
+            assembly (mewarpx.Assembly object): The assembly object in which
+                particles are scraped.
+            species_list (list of species): Species for which histograms should
+                be recorded. If None, all simulation species will be included.
+            linres_x (int): Number of bins to create along the x-dimension.
+            linres_y (int): Number of bins to create along the y-dimension.
+            plot (bool): If True plot the histograms at each diagnostic step.
+            kwargs: See :class:`mewarpx.diags_store.diag_base.WarpXDiagnostic`
+                for more timing options.
+        """
+        self.assembly = assembly
+        self.plot = plot
+
+        # sanity check
+        if not isinstance(self.assembly, assemblies.ZPlane):
+            raise AttributeError(
+                "ZplanePHistDiag should only be used with a ZPlane assembly."
+            )
+
+        # set sensible linres based on simulation geometry - can override
+        # user given arguments if they don't make sense
+        if mwxrun.dim == 1:
+            logger.warn(
+                "ZPlane particle scraping histograms in 1d are trivial so this "
+                "diagnostic will not be installed."
+            )
             return
+        elif mwxrun.dim == 2:
+            linres_y = 1
+
+        # set up domain and number of bins
+        self.domain = [(mwxrun.xmin, mwxrun.xmax), (mwxrun.ymin, mwxrun.ymax)]
+        self.linres = [linres_x, linres_y]
+
+        # for now we hard code this diagnostic to only save weight but this
+        # can be extended in the future
+        self.quantities = ['x', 'y', 'w']
+
+        super(ZplanePHistDiag, self).__init__(
+            diag_steps=diag_steps, species_list=species_list, **kwargs
+        )
 
-        if self.plot:
-            self.plot_histograms()
+    def _process_scraped_particles(self, scraped_particle_dict):
+        """Function to process the scraped particle data.
 
+        Arguments:
+            scraped_particle_dict (dict): Dictionary containing scraped
+                particle data.
+        """
         for ii, species in enumerate(self.species_list):
-            fileprefix = self.get_fileprefix(species.name)
-            np.save(fileprefix + '.npy', self.Harray[ii,...])
+            idx = np.where(
+                scraped_particle_dict['species_id'] == species.species_number
+            )
+            sample = np.zeros((np.size(idx), 2))
+            sample[:,0] = scraped_particle_dict['x'][idx]
+            sample[:,1] = scraped_particle_dict['y'][idx]
+            H, _ = np.histogramdd(
+                sample=sample, bins=self.bins,
+                weights=scraped_particle_dict['w'][idx]
+            )
+            self.Harray[ii,:,:,0] += H
```

Comment:
```suggestion
            self.Harray[ii,...] += H
```
With the above change correcting `Harray` dimension number, this will also be generic over number of dimensions, whereas this code is only correct for 2D...